src/main/java/org/candlepin/insights/pinhead/PinheadService.java

```diff
@@ -52,13 +59,32 @@ public PinheadService(PinheadApiProperties apiProperties, PinheadApi api,
         this.retryTemplate = retryTemplate;
     }
 
-    public OrgInventory getPageOfConsumers(String orgId, String offset) throws ApiException {
+    /**
+     * Return a page of consumers for the given org using the offset provided and with consumers that have
+     * checked in since the lastCheckinTime.  This function accepts the last check in time as a String
+     * parameter in order to validate the format prior to sending the request over the wire.  The format is
+     * similar to ISO 8601 format but the code on the other end is matching to the regular expression we're
+     * using for validation rather than doing full ISO 8601 parsing.  Therefore, things like timezone
+     * offsets are not acceptable for the request even though they are valid ISO 8601 formats.
+     * @param orgId organization id
+     * @param offset offset in the listing of results
+     * @param lastCheckinTime in the form YYYY-MM-DDTHH-MM-SSZ (e.g. 2012-12-18T23:56:23Z).
+     * @return an OrgInventory object with the relevant results
+     */
+    public OrgInventory getPageOfConsumers(String orgId, String offset,
+        @Pattern(regexp = "^([12]\\d{3}-(0[1-9]|1[0-2])-(0[1-9]|[12]\\d|3[01])T(0[0-9]|1[0-9]|2[1-4]):" +
+        "(0[0-9]|[1-5][0-9]):(0[0-9]|[1-5][0-9]))Z$") String lastCheckinTime) throws ApiException {
```

Comment:
Instead of taking in the `lastCheckinTime` here, can we take in `OffsetDateTime` instead please?

Reply:
I had envisioned that this method would only be used internally and I wanted to use method parameter validation.  I can make it protected and see if it still works.  That way on our side, everyone keeps using `getPageOfConsumers(String orgId, String offset)` which formats the time we want and dispatches it to the `getPageOfConsumers(String orgId, String offset, String lastCheckinTime)` version.

Reply:
Fair enough. I didn't read your code comment closely enough :-)

Reply:
Yep, the unit tests confirm that a `protected` method still runs the validation.  So we can at least conceal the String based regex from external callers and convert to the accepted format internally.